src/librustdoc/clean/utils.rs

```diff
@@ -562,3 +547,18 @@ crate fn has_doc_flag(attrs: ty::Attributes<'_>, flag: Symbol) -> bool {
 ///
 /// Set by `bootstrap::Builder::doc_rust_lang_org_channel` in order to keep tests passing on beta/stable.
 crate const DOC_RUST_LANG_ORG_CHANNEL: &'static str = env!("DOC_RUST_LANG_ORG_CHANNEL");
+
+/// Render a sequence of macro arms in a format suitable for displaying to the user
+/// as part of an item declaration.
+pub(super) fn render_macro_arms(matchers: &[TokenTree]) -> String {
+    matchers
+        .iter()
+        .map(|matcher| format!("    {} => {{ ... }};\n", render_macro_matcher(matcher)))
+        .collect::<String>()
```

Comment:
Rather than allocating a new string for each arm, can you use a for loop and push_str?

Reply:
I changed it to use `writeln!`, which I think should generate code fairly similar to `push_str`, but is more readable.